src/components/panel/panel.js

```diff
@@ -932,6 +934,7 @@ MdPanelRef.prototype.detach = function() {
 
     self._panelContainer.remove();
     self.isAttached = false;
+    self._config.locals = null;
```

Comment:
I'm not sure we want this in detach. By putting it here, the user would need to create a new panel in order to open it again (because you just destroyed the data that is used within the panel). This counters the comment you updated above.

Would it make more sense for us to have a destroy method instead that the user can call? @DerekLouie What do you think is the correct behavior, from a user's point of view?